spatialprofilingtoolbox/cggnn/generate_graphs.py

```diff
@@ -333,15 +329,22 @@ def create_graphs_from_specimen(
         df_target = df
     if df_target.shape[0] > max_cells_to_consider:
         df_target = df_target.sample(max_cells_to_consider)
```

Comment:
If this implementation is fast enough, we could even remove these lines that downsample the number of cells to make the distance computation have a ceiling on the time taken. The KDTree still operates in `n log(n)` time though so it could go for a very long time if left unchecked.